lib/Traits/TraitsClasses/TimeVaryingTraits.ts

```diff
@@ -71,4 +71,34 @@ export default class TimeVaryingTraits extends mixTraits(CatalogMemberTraits) {
     type: "string"
   })
   dateFormat?: string;
+
+  @primitiveTrait({
+    name: "Time Interval",
+    description:
+      "Specify a time interval from the current time (forward or backward, depending on `Is Forward`) that this dataset is available.",
+    type: "number"
+  })
+  interval?: number;
+
+  @primitiveTrait({
+    name: "Time Unit",
+    description:
+      "The time unit for the `Time Interval`. Valid values are <br/>" +
+      " * `year` <br/>" +
+      " * `month` <br/>" +
+      " * `day` <br/>" +
+      " * `hour` <br/>" +
+      " * `minute` <br/>" +
+      " * `second` <br/>",
+    type: "string"
+  })
+  timeUnit?: string;
+
+  @primitiveTrait({
+    name: "Is Forward",
+    description:
+      "If true, the interval is current time forward. Otherwise backward. Default to true (forward).",
```

Comment:
I think these optional time window traits are still useful for other time based data type. Don't you think so?

Reply:
I think it is better to not add `Traits` to items unless they are implemented. By adding these to `TimeTraits`, it means they will show up in docs.terria.io for most Catalog Items - even though they are only functional for `ArcGisMapServerCatalogItem`

Reply:
This is a good point. How about putting them into a `TimeWindowTraits` to be used by any types that have its implementation?

Reply:
Agree with @nf-s. 

I think it is best to define the traits in `ArcGisMapServerCatalogItemTraits` for now - mainly because you never know when or if we are going to implement time window for other time varying items. 

When we get to that point it is pretty easy to pull these traits into a common class like you suggested.